### DISCUSSION TITLE
What functions are used by default MappingSchema to convert string to a mapped class (if no custom converters are defined)?

### DISCUSSION THREAD

**Discussion by alexey-leonovich** [linq2db/linq2db#4675] [Q&A] [answered]
Lets say I have a simple `UserRole` class with a `Parse` method:
```
public class UserRole
{
    internal string name { get; set; }

    public static UserRole Parse(string value)
    {
        return new UserRole { name = value };
    }
}
```
and `User` class that has `UserRole` property stored in database as string:
```
[Table]
public class User
{
    [Column(DbType = "nvarchar(50)")]
    public UserRole Role { get; set; }
}
```
Looks like default MappingSchema is smart enough to utilize that method if I don't specify my own string to UserRole converter (with `schema.SetConverter<string, UserRole>( ... )`). The following code successfully discovers and calls `UserRole.Parse("SampleName")`:
```
var schema = new MappingSchema();
var defaultConverter = schema.GetConverter<string, UserRole>();
if (defaultConverter != null)
{
    //equals to UserRole.Parse("SampleName")
    var convertedRole = defaultConverter("SampleName");
}
```
I've took a quick look at linq2db sources and found [that lines](https://github.com/linq2db/linq2db/blob/0c99d98c912ae812657c89ae90a5ccf0e6436e07/Source/LinqToDB/Common/ConvertBuilder.cs#L525-L534) where that heuristics stands. But I guess there are still more places and maybe for other primitive DbTypes (byte, int, DateTime etc..). Is that documented somewhere (function names that will work by default if founded in mapped class)? Or may there is some standard where such function names are enumerated? Thank you.

**Comment by MaceWindu** [comment, marked as answer]
This is the only place like that. That logic was never documented, but I would say it is possible to rely on this behavior as we hardly ever remove support for it.
